src/util/types.ts

```diff
@@ -56,3 +59,50 @@ export const getTableColumns = <
 
   return columns;
 };
+
+/**
+ * Codec to represent built-in `Map` type.
+ *
+ * Be warned that `JSON.stringify()` and `JSON.parse()` don't support `Map` and this can
+ * lead to unexpected behavior if you use this codec in serialization/deserialization.
+ *
+ * ```ts
+ * // Will give you an empty object in string instead of serialized `Map`
+ * JSON.stringify(new Map([[1, 'value']])); // Produces '{}'
+ * ```
+ */
+export const map = <K, V>(
+  domain: t.Type<K>,
+  codomain: t.Type<V>,
+  name = `Map<${domain.name}, ${codomain.name}>`
+): t.Type<Map<K, V>> => {
+  return new t.Type(
+    name,
+    (u): u is Map<K, V> => u instanceof Map,
```

Comment:
`instanceof` check is not enough. It should validate same as validator of `.decode`
```suggestion
    (u): u is Map<K, V> =>
      u instanceof Map &&
      [...u.entries()].every(
        ([key, value]) => domain.is(key) && codomain.is(value)
      ),
```